cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/artifact/ArtifactRepository.java

```diff
@@ -463,17 +463,17 @@ public ArtifactDetail addArtifact(final Id.Artifact artifactId, final File artif
 
     parentArtifacts = parentArtifacts == null ? Collections.<ArtifactRange>emptySet() : parentArtifacts;
     CloseableClassLoader parentClassLoader = null;
-    NamespacedImpersonator namespacedImpersonator = new NamespacedImpersonator(artifactId.getNamespace().toEntityId(),
-                                                                               impersonator);
+    EntityImpersonator entityImpersonator = new EntityImpersonator(artifactId.getNamespace().toEntityId(),
```

Comment:
Yup! hence NamespacedImpersonator -> EntityImpersonator. Added artifacts to supported entity types.